src/theory/sets/theory_sets_rewriter.cpp

```diff
@@ -617,6 +604,22 @@ RewriteResponse TheorySetsRewriter::postRewrite(TNode node) {
   return RewriteResponse(REWRITE_DONE, node);
 }
 
+Node TheorySetsRewriter::rewriteMembershipBinaryOp(const Node& node)
+{
+  NodeManager* nm = nodeManager();
+  std::vector<Node> children;
+  for (size_t i = 0, nchild = node[1].getNumChildren(); i < nchild; i++)
+  {
+  Node nc = nm->mkNode(Kind::SET_MEMBER, node[0], node[1][i]);
+  if (node[1].getKind() == Kind::SET_MINUS && i == 1)
+  {
+    nc = nc.negate();
+  }
+  children.push_back(nc);
+  }
+  return nm->mkNode(node[1].getKind() == Kind::SET_UNION ? Kind::OR : Kind::AND,
```

Comment:
This was the original code, but I prefer to separate the rewriting rule for each operator differently. It is hard to keep track of the cases.

Reply:
Unfortunately this would complicate things quite a bit since it would require splitting cases for proof rules, inference identifiers, and so on. It would also lead to code duplication.